bundles/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/device/DeviceBase.java

```diff
@@ -79,48 +78,39 @@ public void onNotify(String response) {
         boolean needNotify = true;
         JsonObject message = new JsonParser().parse(response).getAsJsonObject();
         try {
-            String updateProp = "";
             if (message.has("method")) {
                 String method = message.get("method").toString().replace("\"", "");
                 if (method.equals("props")) {// Property notify
                     String params = message.get("params").toString();
                     JsonObject propsObject = new JsonParser().parse(params).getAsJsonObject();
-                    Set<Entry<String, JsonElement>> props = propsObject.entrySet();
-                    Iterator<Entry<String, JsonElement>> iterator = props.iterator();
-                    while (iterator.hasNext()) {
-                        Entry<String, JsonElement> prop = iterator.next();
-                        if (prop.getKey().equals("power")) {
-                            updateProp += " power";
+                    for (Entry<String, JsonElement> prop : propsObject.entrySet()) {
+                        final YeelightDeviceProperty property = YeelightDeviceProperty.fromString(prop.getKey());
+                        if (YeelightDeviceProperty.POWER.equals(property)) {
```

Comment:
Wouldn't it be possible to use `switch (property)`  here? Since Java 7 we can switch on strings.

Reply:
I've changed the if/else block to a switch statement.